tsdb/head.go

```diff
@@ -523,7 +523,7 @@ func (h *Head) Init(minValidTime int64) (err error) {
 	h.startWALReplayStatus(startFrom, endAt)
 
 	multiRef := map[uint64]uint64{}
-	if err == nil {
+	if err == nil && startFrom >= snapIdx {
```

Comment:
Weird thought: Since we set `startFrom` to -1 as a placeholder, could there be a problem if someone runs Prometheus with the system clock set to a time before 1970-01-01?

Reply:
Would `math.MinInt` work as a placeholder?

Reply:
`startFrom` is the file index/number, so it will never be negative here :)

Reply:
OK, cool.